proposals/003-metric-improvements.md

```diff
@@ -0,0 +1,269 @@
+|                                                     |              |                                                             |                                                                                                                                    |
+| kroxylicious_filter_to_broker_request_size_bytes    | Distribution | virtual_cluster, node_id, api_key, api_version, filter_name | Records the message size of every **filter-initiated request** goes from the proxy to the upstream (broker).                       |
+| kroxylicious_broker_to_fiter_response_size_bytes    | Distribution | virtual_cluster, node_id, api_key, api_version, filter_name | Records the message size of every **response** for a **filter-initiated request** arrives at the proxy from the upstream (broker). |
+|                                                     |              |                                                             |                                                                                                                                    |
+| kroxylicious_filter_to_broker_request_transit_time  | Distribution | virtual_cluster, node_id, api_key, api_version, filter_name | Records the time taken for each **filter-initiated request** to transit the proxy                                                  |
+| kroxylicious_client_to_broker_response_transit_time | Distribution | virtual_cluster, node_id, api_key, api_version, filter_name | Records the time taken for each **response** for a **filter-initiated request** to transit the proxy                               |
+
+
+The following metrics wil have changes to their labels.
+
+| Metric                                    | Type    | Labels                   | Deprecated Labels | Description                                                                         |
+|-------------------------------------------|---------|--------------------------|-------------------|-------------------------------------------------------------------------------------|
+| kroxylicious_downstream_connections_total | Counter | virtual_cluster, node_id | virtualCluster    | Incremented by one every time a request comes from/response goes to the downstream. |
+| kroxylicious_downstream_error_total       | Counter | virtual_cluster, node_id | virtualCluster    | Incremented by one every time a request goes to/response comes from the upstream.   |
+| kroxylicious_upstream_connections_total   | Counter | virtual_cluster, node_id | virtualCluster    | Incremented by one every time a request comes from/response goes to the downstream. |
+| kroxylicious_upstream_error_total         | Counter | virtual_cluster, node_id | virtualCluster    | Records the time taken for the message to traverse the proxy                        |
```

Comment:
the descriptions of these look wrong.

* the `*error_total` descriptions don't mention errors. 
* the `*connections_total` descriptions talk about requests, but I guess it's supposed to count established connections

Reply:
`the *connections_total descriptions talk about requests, but I guess it's supposed to count established connections` 

Iam not sure but doesn't this increment even if the connection fails?(Based of my understanding)

Reply:
Yes, there is a typo (request vs connection) and it is under specified.

It is a `connection_total` so I guess the intent was for it to count all connections, even those that are short lived due to error.  @SamBarker was that the intent?